### PR TITLE
PRO-1405 Swap Screen Refinements

### PR DESCRIPTION
**Pull request by lalitkumarrock**
https://linear.app/pillarproject/issue/PRO-1405/wip-swap-screen-refinements

## Description

- The loading spinner show until all the calculations have been completed
- After Selects a different offer, only that offer is a focus
- Press "see less" to show selected offer in best offer

## How Has This Been Tested?

- Tested on Simulator 14 Pro
- Tested on Android real device

## Screenshots (if appropriate):
https://github.com/pillarwallet/pillarwallet/assets/82652040/6acbbe42-23ee-4c44-85c4-baf71ee5560c

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/screens/Bridge/Exchange-CrossChain/Exchange.tsx [hunk 1]
```diff
@@ -289,6 +293,15 @@ function Exchange({ fetchExchangeTitle }: Props) {
 
     const bestOffer = sortedOffers?.find((offer) => !!offer.feeInfo && offer.provider !== EXCHANGE_PROVIDER.LIFI);
 
+    if (hideAllOffers && selectedProvider) {
+      const selectedOffer = sortedOffers?.find((offer) => !!offer.feeInfo && offer.provider === selectedProvider);
+      if (selectedOffer) {
+        return sortedOffers.sort(function (x, y) {
+          return x.provider === selectedProvider ? -1 : y.provider === selectedProvider ? 1 : 0;
+        });
```
Comment (IAmKio) [inline src/screens/Bridge/Exchange-CrossChain/Exchange.tsx:301]: Just a note that you can use a shorthand return here `=>` to simplify the code as opposed to nested returns.

**Review by IAmKio — APPROVED** [review]
LGTM

File: src/screens/Bridge/Exchange-CrossChain/OfferCard.js [hunk 2]
```diff
@@ -67,6 +67,7 @@ type Props = {
   onFetchSortingOfferInfo?: (offerInfo: ExchangeOffer) => void,
   isSelected?: ?boolean,
   onFeeInfo?: (feeInfo: ?TransactionFeeInfo) => void,
+  onEstimating?: (estimating: boolean) => void,
```
Comment (MayukhSamadder2000) [inline src/screens/Bridge/Exchange-CrossChain/OfferCard.js:70]: I think the type of setLoading isn't the same as whats being used here

File: src/screens/Bridge/Exchange-CrossChain/OfferCard.js [hunk 3]
```diff
@@ -148,7 +156,7 @@ function OfferCard({
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [feeInfo, estimationErrorMessage, isEstimating]);
 
-  if (estimationErrorMessage || !isVisible) {
+  if (estimationErrorMessage || !isVisible || (!isSelected && isEstimating)) {
```
Comment (MayukhSamadder2000) [inline src/screens/Bridge/Exchange-CrossChain/OfferCard.js:159]: Same as above

**Review by MayukhSamadder2000 — CHANGES_REQUESTED** [review]
Loc

File: src/screens/Bridge/Exchange-CrossChain/OfferCard.js (hunk 3 above)
Comment (lalitkumarrock) [reply src/screens/Bridge/Exchange-CrossChain/OfferCard.js:159]: What we need to do here?

File: src/screens/Bridge/Exchange-CrossChain/Exchange.tsx (hunk 1 above)
Comment (lalitkumarrock) [reply src/screens/Bridge/Exchange-CrossChain/Exchange.tsx:301]: Updated it!

File: src/screens/Bridge/Exchange-CrossChain/Exchange.tsx [hunk 4]
```diff
@@ -289,10 +297,17 @@ function Exchange({ fetchExchangeTitle }: Props) {
 
     const bestOffer = sortedOffers?.find((offer) => !!offer.feeInfo && offer.provider !== EXCHANGE_PROVIDER.LIFI);
 
+    if (hideAllOffers && selectedProvider) {
+      const selectedOffer = sortedOffers?.find((offer) => !!offer.feeInfo && offer.provider === selectedProvider);
```
Comment (MayukhSamadder2000) [inline src/screens/Bridge/Exchange-CrossChain/Exchange.tsx:301]: What's the type of sortedOffers? Do we have a defined type or its any?

File: src/screens/Bridge/Exchange-CrossChain/OfferCard.js (hunk 2 above)
Comment (MayukhSamadder2000) [reply src/screens/Bridge/Exchange-CrossChain/OfferCard.js:70]: So the thing is the function type you've define can take any function here  with a Boolean argument, but shouldn't we make it more typed i mean see the type of setLoading and use that here over the generic function type

File: src/screens/Bridge/Exchange-CrossChain/Exchange.tsx (hunk 4 above)
Comment (lalitkumarrock) [reply src/screens/Bridge/Exchange-CrossChain/Exchange.tsx:301]: ExchangeOffer[]. it's defined now

**Review by MayukhSamadder2000 — APPROVED** [review]
Lgtm

**Pull request merged**